src/compile/data/formatparse.ts

```diff
@@ -1,14 +1,34 @@
-import {isUrlData} from '../../data';
 import {DateTime, isDateTime} from '../../datetime';
 import {FieldDef, isCount} from '../../fielddef';
 import {isEqualFilter, isOneOfFilter, isRangeFilter} from '../../filter';
+import * as log from '../../log';
+import {CalculateTransform, FilterTransform, isCalculate, isFilter} from '../../transform';
 import {QUANTITATIVE, TEMPORAL} from '../../type';
 import {Dict, extend, isArray, isNumber, isString, keys} from '../../util';
-
-import {CalculateTransform, FilterTransform, isCalculate, isFilter} from '../../transform';
-import {Model} from './../model';
+import {VgFormulaTransform} from '../../vega.schema';
+import {Model} from '../model';
 import {DataFlowNode} from './dataflow';
 
+
+function parseExpression(field: string, parse: string): string {
+  const f = `datum["${field}"]`;
+  if (parse === 'number') {
+    return `toNumber(${f})`;
+  } else if (parse === 'boolean') {
+    return `toBoolean(${f})`;
+  } else if (parse === 'string') {
+    return `toString(${f})`;
+  } else if (parse === 'date') {
+    return `toDate(${f})`;
+  } else if (parse.indexOf('date:') === 0) {
+    const specifier = parse.slice(6, parse.length - 1);  // specifier is in "" or ''
+    return `timeParse(${f},"${specifier}")`;
```

Comment:
Then we should sue it everywhere (whenever we generate `datum[...]`). Moved to https://github.com/vega/vega-lite/issues/2179